app/src/main/java/io/mrarm/irc/util/SimpleWildcardPattern.java

```diff
@@ -25,7 +26,29 @@ public static Pattern compile(String str) {
         if (str.length() > pi)
             ret.append(Pattern.quote(str.substring(pi)));
         ret.append('$');
-        return Pattern.compile(ret.toString());
+        return ret.toString();
     }
 
+    public static Pattern pattCompile(String str) throws PatternSyntaxException {
+        int l1 = str.length() - 1;
```

Comment:
This code is pure garbage. l1 and j are terrible variable names, in particularly what the hell is j and why does it take values 1 and 0 (that's a boolean) and if it is a boolean don't name it j.

j seems to be something along removeFirstLastChar. l1 is length - 1 both of which are redundant.

This code can be written in a much more readable way such as:
```
private static String parsePattern(String str) {
  if (str.startsWith("/") && str.endsWith("/"))
    return globToRegexp(str.substr(1, str.length() - 2));
  if (str.startsWith("?") && str.endsWith("?"))
    return str.substr(1, str.length() - 2);
  return globToRegexp(str);
}
public static Pattern compile(String str) {
  return Pattern.compile(str);
}
```

Also pattCompile is a weird name, it's OK but the old name was better (compile), don't rename it.

Reply:
I think the "?" + something + "?" way to specify a pattern sounds redundant, but I may be wrong, what software uses it?

Reply:
That's discussed in #295, but in short, the whole "ed" family in Unix (ed, ex, vi, sed) plus awk & perl allow both // and ?? as regex delimiters.
Having 2 is useful so you can write either /?/ or ?/?.

I think this patch inverts the meaning for ?? forcing glob instead of regex, which I'm not happy about.